### PR TITLE
Add support for Firebase Remote Config

### PR DESCRIPTION
**Pull request by trekforever**
### Description

Add support for Firebase Remote Config. This adds 3 new commands, as well as a new target for deploy:

| Command                        | Description                                                                                                  |
| -------------------------------| ------------------------------------------------------------------------------------------------------------ |
| **remoteconfig:get**           | Get a Firebase project's Remote Config template.                                                             |
| **remoteconfig:versions:list** | Get a list of the most recent Firebase Remote Config template versions that have been published.             |
| **remoteconfig:rollback**      | Roll back a project's published Remote Config template to the version provided by `--version_number` flag.   |

Added `remoteconfig` to `firebase:deploy` - use to update and publish a project's Firebase Remote Config template.

Also updated the CHANGELOG and README as well

### CODE REVIEW

**Comment by trekforever** [comment]
@mbleigh - Mind taking a look at this? I can't seem to explicitly request reviews on this

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
 - Replaces `tslint` with `eslint` as the default linter in new Cloud Functions for Firebase directories. (#1663)
 - Fixes bug where scheduled Cloud Functions for Firebase deploys unexpectedly succeeded with invalid Cloud Scheduler configurations. (#2609)
+- Add support for Firebase Remote Config (see README for list of commands)
```
Comment (rosariopfernandes) [inline CHANGELOG.md:3]: Judging based on past changelog entries, I think this should probably say something along the lines of:

```suggestion
- Adds Firebase Remote Config commands: `remoteconfig:get`, `remoteconfig:versions:list`, `remoteconfig:rollback`.
- Adds a `remoteconfig` target for `firebase deploy`.
```

Comment (trekforever) [reply CHANGELOG.md]: Thanks for the suggestion! Updated :)

**Comment by samtstern** [comment]
Since there are ~300 commits here I just want to drive by with this comment: there's a bug in GitHub where "squash and merge" only works if you click with your mouse and not use Cmd+Enter on your keyboard :-)

File: src/commands/remoteconfig-get.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+function checkValidNumber(versionNumber: string): string {
+  if (typeof Number(versionNumber) == "number") {
+    return versionNumber;
+  }
+  return "null";
+}
+
+module.exports = new Command("remoteconfig:get")
+  .description("Get a Firebase project's Remote Config template")
```
Comment (bkendall) [inline src/commands/remoteconfig-get.ts:27]: don't capitalize the descriptions, please: go/firebase-cli-contributor#create-a-file-for-your-command

```suggestion
  .description("get a Firebase project's Remote Config template")
```

File: src/commands/remoteconfig-rollback.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { requireAuth } from "../requireAuth";
+import { rollbackTemplate } from "../remoteconfig/rollback";
+import { requirePermissions } from "../requirePermissions";
+import { getVersions } from "../remoteconfig/versionslist";
+
+import getProjectId = require("../getProjectId");
+
+module.exports = new Command("remoteconfig:rollback")
+  .description(
+    "Roll back a project's published Remote Config template to the one specified by the provided version number"
```
Comment (bkendall) [inline src/commands/remoteconfig-rollback.ts:13]: Again, don't capitalize the description

```suggestion
    "roll back a project's published Remote Config template to the one specified by the provided version number"
```

File: src/commands/remoteconfig-versions-list.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+const tableHead = ["Update User", "Version Number", "Update Time"];
+
+function pushTableContents(table: Table, version: Version): number {
+  return table.push([version?.updateUser?.email, version?.versionNumber, version?.updateTime]);
+}
+
+module.exports = new Command("remoteconfig:versions:list")
+  .description(
+    "Get a list of Remote Config template versions that have been published for a Firebase project"
```
Comment (bkendall) [inline src/commands/remoteconfig-versions-list.ts:19]: :)

```suggestion
    "get a list of Remote Config template versions that have been published for a Firebase project"
```

File: src/deploy/remoteconfig/index.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+"use strict";
```
Comment (bkendall) [inline src/deploy/remoteconfig/index.ts:1]: `use strict` isn't necessary in TS files (applies throughout)

(wondering if there's a lint rule for this... I'll have to look)

**Review by bkendall — APPROVED** [review]
A few tiny style points, but LGTM!

Could you also add an appropriate point of contact to this file internally? Thanks: http://google3/firebase/cli/g3doc/owners.md

**Comment by bkendall** [comment]
I would also 100% "Squash and merge" this. Please, please, please be sure to do so. And you _must_ do it by pressing the button as Sam suggested. Thanks 👍

**Comment by trekforever** [comment]
@bkendall Sent you a CL to update g3doc/owners.md!
Also just noticed that I can't merge since I don't have write access :( so feel free to squash&merge on my behalf!

**Pull request merged**